### PR TITLE
JSAAL-25 user can view historical data charts

### PR DESCRIPTION
**Pull request by ajaxjiang96**

### CODE REVIEW

File: src/redux/stock/helper.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const processMonthlyData = (data, range) => {
+  // we need to group and calculate the avg by months
+  const key2Data = {};
```
Comment (FairyWind) [inline src/redux/stock/helper.js]: The name 'key2Data' and 'key2Avg' is ambiguous, please give them more clear names.

File: src/redux/stock/reducer.js [hunk 2]
```diff
@@ -1,28 +1,75 @@
...
-        error: action.payload,
+        ...state,
+        loadingStockStatsData: false,
+        stockStatsDataError: action.payload,
       };
+    case FETCH_HISTORY_CHART_DATA_START:
+      return {
+        ...state,
+        loadingHistoricalChartData: true,
+        historicalData: [],
```
Comment (FairyWind) [inline src/redux/stock/reducer.js]: Dont't need to init historicalData in this term.

File: src/redux/stock/reducer.js [hunk 3]
```diff
@@ -1,28 +1,75 @@
...
+        historicalDataError: undefined,
+      };
+    case FETCH_HISTORY_CHART_DATA_SUCCESS:
+      return {
+        ...state,
+        loadingHistoricalChartData: false,
+        historicalDataError: undefined,
```
Comment (FairyWind) [inline src/redux/stock/reducer.js]: Already reset historicalDataError in FETCH_HISTORY_CHART_DATA_START, no need to reset again.

File: src/stock-details/Performance.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  } = useSelector((state) => state.stock);
+
+  useEffect(() => {
+    dispatch(stockActions.fetchHistoryChartData(symbol, timeRange));
+  }, [timeRange]);
+
+  useEffect(
+    () => () => dispatch(stockActions.resetStockInfo()),
+    [],
+  );
```
Comment (FairyWind) [inline src/stock-details/Performance.jsx]: Be consist on the same format when using useEffect().

File: src/stock-details/charts/PerformanceChart.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+const chartConfig = {
+  propsForDots: {
+    r: '2',
+  },
+  xLabelsOffset: 100,
+  backgroundColor: '#fff',
+  backgroundGradientFrom: '#fff',
+  backgroundGradientTo: '#fff',
+  decimalPlaces: 2,
+  labelColor: (opacity = 1) => `rgba(100, 100, 100, ${opacity})`,
```
Comment (FairyWind) [inline src/stock-details/charts/PerformanceChart.jsx]: No need to pass 'opacity' variable here.

File: src/stock-details/Performance.jsx (hunk 4 above)
Comment (ajaxjiang96) [reply src/stock-details/Performance.jsx]: following eslint here

File: src/redux/stock/helper.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+const processDailyData = (data) => ({
+  data: data.map((dailyData) => dailyData.close),
+  keys: data.map((dailyData) => dailyData.date.substr(8, 10)),
+});
+
+export default {
+  processChartData: (data, range) => {
+    if (['1y', '2y', '5y', '6m', '3m'].indexOf(range) > -1) {
```
[the remark this answers is not in the document]
Comment (ajaxjiang96) [reply src/redux/stock/helper.js]: > maybe we can go with daily in case of 3 months as well, what do you think?

I'll check if the labels fits

**Pull request merged**